pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +455,151 @@ def _print_Import(self, expr):
+                return ', '.join(['print *', *fargs]) + '\n'
+            separator = self._print_LiteralString(LiteralString(sep))
+            args_formatting = ''
+            args_code = ''
+            for a_f, a_c in zip(fargs_format, fargs):
+                if a_f == 'A' and len(a_c) > 30 and a_c[0] == '\'' and a_c.count('\'') == 2:
+                    chunks = split_long_argument(a_c)
+                    whole_format = ' '.join(['A'] * len(chunks))
+                    arg_concat = ' , '.join('\'{}\''.format(chunk) for chunk in chunks)
+                    args_code += ('' if args_formatting == ''
+                                  else (' , ' + separator + ' , ' if separator != '' else ' , ')) + arg_concat
+                    args_formatting += ('' if args_formatting == ''
+                                        else (' A ' if separator != '' else ' ')) + whole_format
+                else:
+                    args_code += ('' if args_formatting == ''
+                                  else (' , ' + separator + ' , ' if separator != '' else ' , ')) + (a_c if a_c != '' else '\'\'')
+                    args_formatting += ('' if args_formatting == ''
+                                        else (' A ' if separator != '' else ' ')) + a_f
+            if args_formatting == '':
+                return "write(*, '(A)', advance=\"no\") {}\n".format(self._print_LiteralString(LiteralString(end)))
+            if end != '':
+                return "write(*, '({} A)', advance=\"no\") {} , {}\n".format(args_formatting, args_code, self._print_LiteralString(LiteralString(end)))
+            return "write(*, '({})', advance=\"no\") {}\n".format(args_formatting, args_code)
+
+        if len(orig_args) == 0:
+            return formatted_args_to_print(args_format, args, end)
+
+        for i, f in enumerate(orig_args):
             if f.keyword:
                 continue
             else:
                 f = f.value
             if isinstance(f, str):
-                args.append("'{}'".format(f))
+                arg_format, arg = self.get_print_format_and_arg(f)
+                args_format.append(arg_format)
+                args.append(arg)
             elif isinstance(f, PythonTuple):
-                for i in f:
-                    args.append(self._print(i))
+                arg_format, arg = self.get_print_format_and_arg(f)
+                args_format.append(arg_format)
+                args.append(arg)
             elif isinstance(f, InhomogeneousTupleVariable):
-                for i in f:
-                    args.append(self._print(i))
-            elif f.dtype is NativeString() and j != n-1:
-                args.append("{} // ' ' ".format(self._print(f)))
-            else:
+                arg_format, arg = self.get_print_format_and_arg(f)
+                args_format.append(arg_format)
+                args.append(arg)
+            elif f.dtype is NativeString() and f != expr.expr[-1]:
+                args_format.append('A')
```

Comment:
If I do:
```
a = 'hello'
print(a, " is the same as ", a)
```
then this condition will not be evaluated as you expect. You should test `i` not `f`

Reply:
i don't think that check is necessary. I m not sure what it was checking for in the previous code.
Also, this example raised an issue i did not consider.
the output of this file will give:
`program prog_tupp



  implicit none

  character(len=280) :: a

  a = 'hello'
  write(*, '(A A A A A A)', advance="no") a , ' ' , ' is the same as ' , ' ' &
      , a , ACHAR(10)

end program prog_tupp
`
and this will print:
`hello                                                                                                                                                                                                                                                                                     is the same as  hello                                                                                                                                                                                                                                                                                   
`
with lots of extra white spaces due to the fact that the length of a is 280. Shouldn't we assign the exact length of a to get a nicer print?

Reply:
Strings are generally a mess at the moment (see epic #459 ). Fixing them is waaaay beyond the scope of this PR. You can try to improve the situation by using `trim` in the fortran code, but we are aware that this is a big mess. This is part of the reason why the tests never compare whitespace